### PR TITLE
Add glossary of terms to README

### PR DESCRIPTION
**Pull request by kadamwhite**
In an effort to standardize how we refer to different objects within our data model in a future-oriented way, this PR adds a glossary of terms to the README:

## Glossary of Terms

This section clarifies the verbiage and terms used within the application code.

### Attribute

An **attribute** is the object representing a specific type of annotation a user will apply to any given image, such as demographic attributes like "perceived ethnicity" or positional attributes like "the location of the right eye."

Attribute properties:

- **Name**: the human-oriented label of an attribute, such as "Perceived Gender."
- **Type**: the type of data represented by the attribute, *e.g.* a list of multiple-choice options, or a set of coordinates, etc. _Note: This is a database- / data model-oriented value, and has no inherent correspondence to front-end presentation._
- **Options**: the list of accepted values for a multiple-choice image attribute.

Attribute objects have the shape
```json
{
    "name": "Attribute Name",
    "type": "type-of-data",
    "options": ["list", "of", "accepted", "values", "for", "multiple", "choice", "attributes"]
}
```

### Annotation

An **annotation** is the object representing the submitted value for a particular attribute.

Annotation properties:

- **Name**: the name of the image attribute annotated, such as "Perceived Ethnicity".
- **Value**: the value with which the image is annotated, such as "Asian" or "Black."

Annotation objects have the shape
```json
{
    "name": "Attribute Name",
    "value": "selected-value"
}
```

### Image Annotation

An **image annotation** is an object associating one or more annotations with the specific image to which they were applied. It contains an array of annotations, and the ID of their associated image.

Image annotation objects have the shape:
```json
{
    "id": "numeric-image-id",
    "annotations": [{
        "name": "Attribute Name",
        "value": "selected-value"
    }, {
        "name": "Another Attribute Name",
        "value": "selected-value"
    }]
}
```

### CODE REVIEW

**Comment by kadamwhite** [comment]
Technically that last bit should read "An image annotation object associates one or more annotations with a specific image ID, with the shape:"

File: README.md [hunk 1]
```diff
@@ -101,3 +101,49 @@ $ npm run migrate:up
...
+### Annotation
+
+An **image annotation** is the object representing the value of a specific attribute a user applied to a specific image.
+
+Image annotation properties:
+
+- **Name**: the name of the image attribute annotated, such as "Perceived Ethnicity".
+- **Value**: the value with which the image is annotated, such as "Asian" or "Black."
+
+Annotation objects have the shape
```
Comment (gnarf) [inline README.md]: Hrm, this is like the "annotatedImages" model, which contains "annotation" objects inside "annotations" right?

Comment (kadamwhite) [inline README.md]: Yep, updating to this:

### Annotation

An **annotation** is the object representing the submitted value for a particular attribute.

Annotation properties:

- **Name**: the name of the image attribute annotated, such as "Perceived Ethnicity".
- **Value**: the value with which the image is annotated, such as "Asian" or "Black."

Annotation objects have the shape
```json
{
    "name": "Attribute Name",
    "value": "selected-value"
}
```

### Image Annotation
An **image annotation** is an object associating one or more annotations with the specific image to which they were applied. It contains an array of annotations, and the ID of their associated image.

Image annotation objects have the shape:
```json
{
    "id": "numeric-image-id",
    "annotations": [{
        "name": "Attribute Name",
        "value": "selected-value"
    }, {
        "name": "Another Attribute Name",
        "value": "selected-value"
    }]
}
```

Comment (gnarf) [inline README.md]: An a AnnotatedWorkload contains a workload id and multiple AnnotatedImages, etc.

The workload fetched from the server would just have id and Images...

Comment (gnarf) [inline README.md]: nit: use `### ImageAnnotation` so that we can ``[`ImageAnnotation`][#ImageAnnotation]`` to refer back to it

Comment (kadamwhite) [inline README.md]: Yeah, the workload fetched from the server doesn't actually contain anything annotation-specific, it's got a separate set of image attributes

**Comment by kadamwhite** [comment]
@gnarf Updated the description with the latest pushed README, based on your feedback

File: README.md (hunk 1 above)
Comment (gnarf) [inline README.md]: So we get a `Workload` and post an `AnnotatedWorkload`

**Comment by boazsender** [comment]
my problem with `attributes` is that it's a generic concept in almost all software. It's so generic, that I think it makes it hard to reason about. Kind of like how `node`s in drupal run away from you.

If we think about this more specifically, what we're talking about is what what to name the things this system can possibly describe about the subject of an image, ie, what can be annotated. What is a more domain specific name for that concept?

**Comment by kadamwhite** [comment]
> What is a more domain specific name for that concept?

This is what I've been struggling with today, and I'm afraid I don't have a good suggestion. I like attribute, or image attribute, in part _because_ it's generic; but I appreciate your argument, it's a different perspective.

**Comment by gnarf** [comment]
> What is a more domain specific name for that concept?

I'm not sure there is one, it is quite generic because it deals with many possible datas.

There aren't many words that describe the shared concepts of locations on the face, regions of the face, and demographics, but `attribute` does have the advantage of making sense for all of them in an abstract way.

We could split from the idea of using a shared model for all three, and call them demographics, regions, and points - but attributes does seem fairly future-kind to other types of annotations

**Comment by kadamwhite** [comment]
Going to land this to unblock myself. @gnarf If you're going to handle the backend/endpoint updates, could you make an issue to track that?

**Comment by boazsender** [comment]
What about
`annotations`<-- types of annotations
`imageAnnotation` <-- single annotation
`imageAnnotations` <--- group of annotations on an image

**Comment by gnarf** [comment]
I find `annotations` being used as the "types" to be less intuitive, as an annotation is actually a mark on a thing, not a classification of the types of marks

**Comment by kadamwhite** [comment]
I agree with @gnarf's comment, I struggle with "annotation" being the umbrella term for the attribute/type of data with which the image is annotated

**Comment by boazsender** [comment]
What do you think of `annotatableAttributes`?

**Comment by gnarf** [comment]
> `annotatableAttributes`

`annotation_attribute` makes some sense as a DB name IMO

`GET /api/annotation/attributes` makes some sense

we still get `var attribute` for shorthand.

The data formats for `AnnotatedImage` can still have `{ annotations: [] }`

Maybe instead of `name` in our `Annotation` format though we should call it `{ attribute, value }` then?

**Comment by boazsender** [comment]
OK, I'm +1 to this approach. Roll out the refactor how ever you see fit, @gnarf @kadamwhite 

Please take into account this PR as you do: https://github.com/bocoup/image-annotator/pull/78

And keep master working.

In addition to the commits on PR #78, I was planning to add a commit for logic to set a new truth in the `known` based on some arbitrary criteria of a "annotatable attribute". Or maybe the first pass is '4 different annotators agree'.

**Comment by gnarf** [comment]
@boazsender When were you thinking about working on that change? I'll have some time to really go heads down on backend stuff at around 3 today, so if we can get close to done with your change, we can merge that, and then I can get to work on the backend swap -- the frontend commit should be small enough to not be a huge diff.

**Comment by kadamwhite** [comment]
Per Boaz' comments in stand today I don't think he'll have much time to work on that before 3

**Comment by boazsender** [comment]
Not sure. I'll try to work on it before 3, but I'm not sure if I'll get to it. I have a lot of meetings today.

**Comment by kadamwhite** [comment]
@gnarf 

> Maybe instead of `name` in our Annotation format though we should call it `{ attribute, value }` then?

I'm fine with either, let me know what you'd like to go with and I'll make the front-end conform

**Comment by kadamwhite** [comment]
After discussion with @gnarf the reference from an Annotation to an Annotatable Attribute will continue to be called "name"

**Comment by kadamwhite** [comment]
Front-end updated in #85

**Pull request merged**